hubspot3/tickets.py

```diff
@@ -78,3 +90,181 @@ def get_all(self, limit: int = -1, **options) -> list:
             offset = batch["offset"]
 
         return output if not limited else output[:limit]
+
+    def _get_batch(self, ids: List[int], extra_properties: Union[List[str], str] = None,
+                   with_history: bool = False) -> Dict[str, dict]:
+        """given a batch of vids, get more of their info"""
+        # default properties to fetch
+        properties = set(self.__default_batch_properties)
+
+        # append extras if they exist
+        if extra_properties:
+            if isinstance(extra_properties, list):
+                properties.update(extra_properties)
+            if isinstance(extra_properties, str):
+                properties.add(extra_properties)
+        if with_history:
+            property_name = "propertiesWithHistory"
+        else:
+            property_name = "properties"
+        params = {}
+        params[property_name] = list(properties)
+        params['includeDeletes'] = True
+        # run the ids as a list of 100
+        batch = {}
+        while len(ids) > 0:
+            partial_ids = ids[:100]
+            ids = ids[100:]
```

Comment:
I think it is better to not redefine names, especially function parameters. When reading the while condition I thought you were mutating the list passed by the caller.